tools/webpack/packages.js

```diff
@@ -0,0 +1,175 @@
+/**
+ * External dependencies
+ */
+const LiveReloadPlugin = require( 'webpack-livereload-plugin' );
+const CopyWebpackPlugin = require( 'copy-webpack-plugin' );
+const postcss = require( 'postcss' );
+
+const { join } = require( 'path' );
+
+/**
+ * WordPress dependencies
+ */
+const LibraryExportDefaultPlugin = require( '@wordpress/library-export-default-webpack-plugin' );
+
+const baseDir = join( __dirname, '../../' );
+
+/**
+ * Given a string, returns a new string with dash separators converedd to
+ * camel-case equivalent. This is not as aggressive as `_.camelCase` in
+ * converting to uppercase, where Lodash will convert letters following
+ * numbers.
+ *
+ * @param {string} string Input dash-delimited string.
+ *
+ * @return {string} Camel-cased string.
+ */
+function camelCaseDash( string ) {
+	return string.replace(
+		/-([a-z])/g,
+		( match, letter ) => letter.toUpperCase()
+	);
+}
+
+module.exports = function( env = { environment: 'production' } ) {
+	const mode = env.environment;
+	const suffix = mode === 'production' ? '.min': '';
+
+	const gutenbergPackages = [
+		'a11y',
+		'api-fetch',
+		'autop',
+		'blob',
+		'blocks',
+//		'block-library', // Not on npm yet.
+		'block-serialization-default-parser',
+//		'block-serialization-spec-parser', // No build-module folder.
+		'components',
+		'compose',
+		'core-data',
+		'data',
+		'date',
+		'deprecated',
+		'dom',
+		'dom-ready',
+//		'edit-post', // Not on npm yet.
+		'editor',
+		'element',
+//		'escape-html', // Not on npm yet.
+		'hooks',
+		'html-entities',
+		'i18n',
+//		'is-shallow-equal', // No build-module folder.
+		'keycodes',
+//		'list-reusable-blocks', // Not on npm yet.
+		'nux',
+		'plugins',
+		'redux-routine',
+//		'rich-text', // Not on npm yet.
+		'shortcode',
+		'token-list',
+		'url',
+		'viewport',
+		'wordcount',
+	];
+
+	const packagesStyles = [
+		'nux',
+		'components',
+		'editor',
+	];
+
+	const externals = {
+		react: 'React',
+		'react-dom': 'ReactDOM',
+		tinymce: 'tinymce',
+		moment: 'moment',
+		jquery: 'jQuery',
+		lodash: 'lodash',
+		'lodash-es': 'lodash',
```

Comment:
These externals are mostly downloaded by Gutenberg at build time. For WordPress, we should just add them as npm dependencies and bundle them as scripts (like the WordPress packages). You can see how I did that for lodash in this patch https://core.trac.wordpress.org/ticket/44987

(expect jQuery and tinymce of course which are already there)